completion/plaintex.cwl

```diff
@@ -0,0 +1,4 @@
+\ttfam#*
```

Comment:
why is this moved to a separate file ?

Reply:
These lines are added to `tex.cwl` in pr #783 opened by myself.

1. All four `\xxfam` commands are macros, not primitive, and are never defined in LaTeX format but only in plainTeX format. 
1. Moreover, I find having them in autocompletion makes the autocompletion less convenient. For example when I type `\ttf` and hit TAB, what I expect is to have `\ttfamily` autocompleted, but with `\ttfam` there I must type more characters to distinguish `\ttfam` from `\ttfamily. (I always set autocompletion "all".)
1. There might be more plainTeX macros currently recorded in `tex.cwl`, but these four are the most annoying ones hence I removed them. 

In most cases, texstudio is used to edit a tex file written in LaTeX format (starting with `\documentclass`), so this change is transparent to most users. PlainTeX users can check `plaintex.cwl` in configuration to add support for plainTeX, hence this change would not remove any utilities.

So in short, remove from `tex.cwl` for LaTeX users, and move to separate `plaintex.cwl` for plainTeX users (and a bit of backward compatibility).

Reply:
I am a bit confused why it should be moved from a cwl which is already named tex.cwl.

Reply:
Any particular reason not use "typical" completion ?

Reply:
(Sorry it takes me too long to compose and adjust words and sentences to express my intension.)

> I am a bit confused why it should be moved from a cwl which is already named tex.cwl.

Not sure if I understand your confusion, but I guess this is similar to #751.

Concerning the 3 auto-loaded cwl files (`tex.cwl`, `latex-document.cwl` and `latex-dev.cwl`), what they should provide (and provide only) can be split into two parts:
 - the primitives provided by a tex engine, and
 - the commands (including environments) provided by a tex format.

The most common combination is pdftex engine + latex format, and this is what the most lines in those three cwl files do.
 - To add support for engines other than pdftex, separate files [`xetex.cwl`](https://github.com/texstudio-org/texstudio/blob/master/completion/xetex.cwl) and [`luatex.cwl`](https://github.com/texstudio-org/texstudio/blob/master/completion/luatex.cwl) were added years ago. And these two cwl files are not auto-loaded.
 - To add support for formats other than latex,  my decision in pr #783 was to add lines for plainTeX format into `tex.cwl`. But I found this inconvenient, hence in this pr I started to move plainTeX-only lines to separate file `plaintex.cwl`.

This change makes handling for different formats the same for the handling for different engines.

BTW, in practice it is very very rare that one `.tex` file is deliberately made compilable in different formats (compilable both by executable `pdftex` and `pdflatex`), but it is not such rare that one `.tex` file is compilable in different engines. Therefore recording commands for different formats in different files dose, in most cases, no harm.

> Any particular reason not use "typical" completion ?

I need to type in macros marked `#*` in cwl (used for completion only in with the "all" tab) too often. Perhaps I am too far away from typical latex users.

Reply:
Are you aware that txs counts completion usage and that rare commands appear in typical if they are completed a few times ?

I see that you yourself added these commands ... Are you sure that not even more can be moved ? Default should contain stuff which most users need ...

Reply:
> Are you aware that txs counts completion usage and that rare commands appear in typical if they are completed a few times ?

Thanks. I will have a try.

> I see that you yourself added these commands ... Are you sure that not even more can be moved ?

There are definitely more can be moved, though currently this pr just moves the most annoying ones. I've found a recent doc ([`tex-nutshell.pdf`](http://mirrors.ctan.org/info/tex-nutshell/tex-nutshell.pdf)) which has a summary of plainTeX macros in sec. 14.

If we both agree that plainTeX-only macros can be moved from `tex.cwl` to `plaintex.cwl`, I can do the rest moving things in one or two days. This way related changes are kept in single pr.

> Default should contain stuff which most users need ...

Most users don't use plainTeX.

Reply:
Agreed

Reply:
> [...] I can do the rest moving things in one or two days. This way related changes are kept in single pr.

The moving work has been done in #1203, so I'll mark this conversation as resolved.